go-controller/pkg/ovn/egressip_test.go

```diff
@@ -2787,6 +2811,9 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations", func() {
 					&v1.NamespaceList{
 						Items: []v1.Namespace{*egressNamespace},
 					},
+					&v1.PodList{
```

Comment:
This doesn't seem correct to me. The test is testing if ovn-kubernetes is able to delete any egress IP setup for pods that do not exist anymore when ovn-kubernetes starts. This line right here actually does the inverse, it adds the pod to the initial kube state, simulating that the pod does exist when ovn-kubernetes starts.

Reply:
I had to add this pod in order to exercise the code that deletes egress ips.
W/out that, the test would not get to exercise `syncEgressIPs` because `generatePodIPCacheForEgressIP` would return an empty cache.

Reply:
This is what I am looking for when running that test:
```
I0106 13:22:59.900875 2530760 egressip.go:874] syncStaleEgressReroutePolicy will delete egressip: &{0c99c791-ca28-4db0-9566-41f009f24dc9 reroute map[name:egressip] ip.src == 10.128.3.8 <nil> [] map[] 100}
```

Reply:
You are right. I think I made the test confusing and less useful.
I will undo this change; especially because the router_test added already exercise the fix for this bug.